### PR TITLE
Improve documentation

### LINKED ISSUE #65: Some docstrings are not correctly rendered in the documentation
- [x] `Missing docstring for ClimaDiagnostics.DiagnosticVariables.long_name. Check Documenter's build log for details.`
- [x] file:///home/sbozzolo/repos/ClimaDiagnostics.jl/docs/build/writers.html
- [x] broken link in docs "User Guide" page, [Schedules section](https://clima.github.io/ClimaDiagnostics.jl/dev/user_guide/#Schedules) - link to `Schedules` near bottom of section

### PR DESCRIPTION
**Pull request by Sbozzolo**
Closes #65

### CODE REVIEW

**Comment by Sbozzolo** [comment]
@juliasloan25 could you have a look and let me know if I addressed #65?

File: docs/src/developer_guide.md [hunk 1]
```diff
@@ -293,7 +338,7 @@ like the following:
 ```julia
 import ClimaDiagnostics: average_pre_output_hook!, HDF5Writer, NetCDFWriter, ScheduledDiagnostic
 
-function parse_yaml(parsed_args, target_space)
+function parse_yaml(parsed_args, source_space)
```
Comment (juliasloan25) [inline docs/src/developer_guide.md:341]: src/netcdf_writer_coordinates.jl still has a function `target_coordinates`, which maybe should be changed to `source_coordinates` now? Also, the docstring for that function doesn't match the function - it has an extra `!` and type annotation

**Review by juliasloan25 — APPROVED** [review]
This looks good! The section for what to do if you don't have an integrator is helpful too

File: docs/src/developer_guide.md (hunk 1 above)
Comment (Sbozzolo) [reply docs/src/developer_guide.md:341]: It is `target_coordinates` because these are the coordinates in the target Cartesian space

**Pull request merged**